caikit/core/data_model/base.py

```diff
@@ -477,7 +477,20 @@ def __setattr__(self, name, val):
 
         # If attempting to set one of the named fields or a oneof, instead set
         # the private version of the attribute.
-        if name in cls.fields or name in cls._fields_oneofs_map:
+        if name in cls.fields:
+            field_descriptor = cls._proto_class.DESCRIPTOR.fields_by_name[name]
+            # for bytes, convert to bytes if not already
+            if field_descriptor.type == field_descriptor.TYPE_BYTES:
+                error.value_check(
+                    "<COR13072712E>",
+                    isinstance(val, (bytes, str)),
+                    "{} must be string or bytes",
+                    val,
+                )
+                if isinstance(val, str):
```

Comment:
I guess technically we can convert other types to bytes as well

Reply:
Right, this is my concern with going this route. You could easily cast an `int` to `bytes` too